src/mavsdk/plugins/gimbal/gimbal_protocol_v2.cpp

```diff
@@ -18,6 +18,15 @@ GimbalProtocolV2::GimbalProtocolV2(
     _gimbal_manager_compid(gimbal_manager_compid)
 {}
 
+GimbalProtocolV2::~GimbalProtocolV2()
+{
+    if (_is_mavlink_manager_status_registered) {
+        _is_mavlink_manager_status_registered = false;
+
+        _system_impl.unregister_mavlink_message_handler(MAVLINK_MSG_ID_GIMBAL_MANAGER_STATUS, this);
```

Comment:
Not sure if it matters, but I'm always more comfortable setting the boolean after the fact (i.e. inverting those lines). So that it reads "now that I unregistered, I update the boolean that holds the registration state".

Probably just a nitpick though.

Reply:
The advantage of that way is that the check and set are right after each other and less likely to drift apart from each other and get lost.

Best case, it should not be a two step approach but one. Actually, we don't need this variable. We can just unregister and if we haven't registered it's silently ignored...